### PR TITLE
alanlima/c03-aws03

### PR DESCRIPTION
**Pull request by alanlima**
# Description

Exercise (e.g. C01-E01): **c03-aws03**

Student Name: **Alan Lima**

> Used Route 53 to create private and private hosted zone to the ALB.

## Type of change

- [x] Exercise Submission (one exercise per PR)
  - [x] This PR is just for **one** exercise of a class
  - [x] PR name follows the pattern `<github-username>/<exercise-number>`
  - [x] Added exercise files inside folder `/classes/<class- number>/exercises/<exercise-number>/<github-username>/`
	- [x] Review [exercise submission steps](./README.md#Exercises)
- [ ] Class content (instructors/admins)
- [ ] Documentation update
- [ ] Repo management

### CODE REVIEW

File: classes/03class/exercises/c03-aws03/alanlima/README.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    [ec2-user@ip-172-16-10-104 ~]$ curl c03aws03.devops.rocks
+    ip-172-16-10-117.ap-southeast-2.compute.internal
+    
+    [ec2-user@ip-172-16-10-104 ~]$ curl c03aws03.devops.rocks
+    ip-172-16-10-117.ap-southeast-2.compute.internal
+    ```
+
+- Post any findings, how option you decided to go with, records created, domain created, and anything else that you find interesting.
+
+    - I was going to do the public hosted zone, but it took a while until I receive the e-mail for the free domain registration. Then, I started the private, make it works, got excited, received a verification e-mail, did the public zone as well 😅
```
Comment (denstorti) [inline classes/03class/exercises/c03-aws03/alanlima/README.md:118]: Nice! Yes, it takes some time to get the email and then several hours for the DNS name to propagate for the public one.

File: classes/03class/exercises/c03-aws03/alanlima/README.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    - AWS will set the hosted zone to private when VPC is specified, even if you set PrivateZone=False. I needed to recreate the route a couple of times until I release that. The CLI didn't give any warning. 😢
+    - I found the docs to update the recordset not as clear as the other docs. I could not find exactly how I should create the A record to link to the ALB. I ended up creating it on the console and listed the resources to check how it was created.
```
Comment (denstorti) [inline classes/03class/exercises/c03-aws03/alanlima/README.md:120]: It is a special linking in AWS called `alias`. So, in fact, it is an A record (supposed to translate a name to a set of IPs) that points to a name. 👍

File: classes/03class/exercises/c03-aws03/alanlima/README.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    - I would expect some small delay until the route 53 propagate the changes, but as soon as I set it, I was able to connect to my private hosted zone. Impressive 👌👌.
+    - Overall, I found it quite easy to set up a private hosted zone. When I started, I thought that I would need to execute a thousand commands.
+    - Forgot to set the key pair to my instances so I needed to recreate the launch configuration, ASG and update the LB register again. 🙄 *I am getting good in recreating things 🤣🤣*
```
Comment (denstorti) [inline classes/03class/exercises/c03-aws03/alanlima/README.md:123]: experience! 😆 

Interesting that now ASG supports `launch templates` (along with `launch configurations`). Launch templates are user for a long time for EC2 instances, it is an easy way to template your instance and launch it whenever you want.  Now you can reuse the same template for either EC2 instances or ASGs.

File: classes/03class/exercises/c03-aws03/alanlima/README.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    - I cannot set CNAME record with AliasTarget; the connection will be refused. But it works as expected for the standard way with TTL and string value to DNS alias.
```
Comment (denstorti) [inline classes/03class/exercises/c03-aws03/alanlima/README.md:124]: Good test.

CNAME translates a name to another name. So basically you are telling R53 to map your name to the ALB DNS name. 

** The problem with this approach is that AWS does not guarantee that the DNS name of the ALB won't change, so better to use `alias`. **

File: classes/03class/exercises/c03-aws03/alanlima/README.md (hunk 1 above)
Comment (alanlima) [reply classes/03class/exercises/c03-aws03/alanlima/README.md:118]: Actually, the DNS propagation was fast. It took only a few seconds between setting the namespaces and be able to use the domain.

File: classes/03class/exercises/c03-aws03/alanlima/README.md (hunk 2 above)
Comment (alanlima) [reply classes/03class/exercises/c03-aws03/alanlima/README.md:120]: Yeah, I read that in the console where he said that it is a A record (after I failed miserably to set it up via CLI 😑😑)

File: classes/03class/exercises/c03-aws03/alanlima/README.md (hunk 3 above)
Comment (alanlima) [reply classes/03class/exercises/c03-aws03/alanlima/README.md:123]: Interesting, I am going to research more about it. I actually was creating some copy with original launch configurations for some testings and I was renaming it to v2, v3, and so on. Launch templates has versions control build-in. Cool 🤗🤗

File: classes/03class/exercises/c03-aws03/alanlima/README.md (hunk 4 above)
Comment (alanlima) [reply classes/03class/exercises/c03-aws03/alanlima/README.md:124]: thanks, I will take note on that.

**Pull request merged**